src/utils/HTTPUtils.ts

```diff
@@ -157,6 +157,47 @@ class HTTPUtils {
 
         return result;
     }
+
+    static parseArgs(a: any, b: any, c: any) {
```

Comment:
I think, this might be easier to read and refactor the code later by handling each case individually in if/else statements:
```js
    static parseArgs(a: any, b: any, c: any) {
        let url = a;
        let options = b;
        let callback = c;

        // Check whether first arg is URL 
        if (!(['string', 'URL'].includes(typeof a))) {
          url = a;
          if (typeof url === 'URL' || (url.constructor && url.constructor.name === 'URL')) {
              url.path = url.pathname;
          }
          if (typeof b === 'object' && !c) {
            // handling case of "got.post(url, options)"
            options = b;
            callback = undefined;
          } else if (typeof b === 'function' && !c) {
            // handling case of "request(url, callback)"
            options = undefined;
            callback = b;
          }
        } else if (typeof a === 'object' && typeof b === 'function' && !c) {
            // handling case of "request(options, callback)"
            url = undefined;
            options = a;
            callback = b;
        } 
        
        return { url, options, callback };
    }
```